airflow/providers/amazon/aws/hooks/redshift_sql.py

```diff
@@ -62,6 +80,9 @@ def _get_conn_params(self) -> dict[str, str | int]:
 
         conn_params: dict[str, str | int] = {}
 
+        if conn.extra_dejson.get("iam", False):
```

Comment:
I'm worry that `redshift-connector` also might use same connection parameter, see list available parameters: https://github.com/aws/amazon-redshift-python-driver#connection-parameters

As result it might broke someone connections

Reply:
@IAL32 thoughts on this?

Reply:
I am not sure if this would break any connections. @o-nikolas I will test this week and come back with findings.